XamarinCommunityToolkit/Views/SideMenuView.shared.cs

```diff
@@ -0,0 +1,552 @@
+            = BindableProperty.Create(nameof(CancelVerticalGestureThreshold), typeof(double), typeof(SideMenuView), 1.0);
+
+        public static readonly BindableProperty AllowInterceptGestureProperty
+            = BindableProperty.Create(nameof(AllowInterceptGesture), typeof(bool), typeof(SideMenuView), false);
+
+        public static readonly BindableProperty StateProperty
+            = BindableProperty.Create(nameof(State), typeof(SideMenuState), typeof(SideMenuView), SideMenuState.Default, BindingMode.TwoWay, propertyChanged: OnStatePropertyChanged);
+
+        public static readonly BindableProperty CurrentGestureStateProperty
+            = BindableProperty.Create(nameof(CurrentGestureState), typeof(SideMenuState), typeof(SideMenuView), SideMenuState.Default, BindingMode.OneWayToSource);
+
+        public static readonly BindableProperty PositionProperty
+            = BindableProperty.CreateAttached(nameof(GetPosition), typeof(SideMenuPosition), typeof(SideMenuView), SideMenuPosition.None);
+
+        public static readonly BindableProperty MenuWidthPercentageProperty
+            = BindableProperty.CreateAttached(nameof(GetMenuWidthPercentage), typeof(double), typeof(SideMenuView), -1.0);
+
+        public static readonly BindableProperty MenuGestureEnabledProperty
+            = BindableProperty.CreateAttached(nameof(GetMenuGestureEnabled), typeof(bool), typeof(SideMenuView), true);
+
+        readonly List<TimeDiffItem> timeDiffItems;
+
+        readonly SideMenuElementCollection children;
+
+        AbsoluteLayout mainLayout;
+
+        readonly View overlayView;
+
+        View mainView;
+
+        View leftMenu;
+
+        View rightMenu;
+
+        View activeMenu;
+
+        View inactiveMenu;
+
+        double zeroDiff;
+
+        bool isGestureStarted;
+
+        bool isGestureDirectionResolved;
+
+        bool isSwipe;
+
+        double previousDiff;
+
+        public SideMenuView()
+        {
+            timeDiffItems = new List<TimeDiffItem>();
+
+            children = new SideMenuElementCollection();
+            children.CollectionChanged += OnChildrenCollectionChanged;
+
+            overlayView = SetupMainViewLayout(new BoxView
+            {
+                InputTransparent = true,
+                GestureRecognizers =
+                {
+                    new TapGestureRecognizer
+                    {
+                        Command = new Command(() => State = SideMenuState.Default)
+                    }
+                }
+            });
+
+            if (Device.RuntimePlatform != Device.Android)
+            {
+                var panGestureRecognizer = new PanGestureRecognizer();
+                panGestureRecognizer.PanUpdated += OnPanUpdated;
+                GestureRecognizers.Add(panGestureRecognizer);
+            }
+
+            ControlTemplate = new ControlTemplate(typeof(AbsoluteLayout));
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
Can we come up with a way to not use these constructions? I'd love to use no (or as less as possible) `EditorBrowsable` if we can